src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectReader.cs

```diff
@@ -943,11 +943,32 @@ internal Type FastBindToType(string assemblyName, string typeName)
             if (entry == null || entry.AssemblyName != assemblyName)
             {
                 Assembly assm = null;
-                try
+                if (_isSimpleAssembly)
                 {
-                    assm = Assembly.Load(new AssemblyName(assemblyName));
+                    try
+                    {
+                        assm = Assembly.Load(assemblyName);
+                    }
+                    catch { }
+
+                    if (assm == null)
```

Comment:
[Per desktop ](https://referencesource.microsoft.com/#mscorlib/system/runtime/serialization/formatters/binary/binaryobjectreader.cs,1428)this should also check `assemblyName != null`